src/main/k8s/production.yml

```diff
@@ -28,6 +28,10 @@ spec:
           requests:
             cpu: 200m
             memory: 512Mi
+      volumes:
```

Comment:
Please check https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-files-from-a-pod for how to use secret inside pod's container. You need 2 things: a volume declaration (that reference the secret itself) and a volume mount to indicate where the volume should be mounted.